core/services/workflow_service/controllers/compute_block_controller.py

```diff
@@ -0,0 +1,219 @@
+from fastapi import HTTPException
+import requests
+import os
+from typing import List, Dict, Optional, Union
+from uuid import UUID
+
+from sqlalchemy.orm import Session
+from utils.database.session_injector import get_database
+from services.workflow_service.models.block import Block
+from services.workflow_service.models.entrypoint import Entrypoint
+from services.workflow_service.models.inputoutput import (
+    InputOutput, InputOutputType
+)
+from services.workflow_service.schemas.compute_block import (
+    UpdateEntrypointDTO, UpdateInputOutputDTO
+)
+
+from scystream.sdk.config import SDKConfig, load_config
+
+TEMP_DIR = "tmp/"
+# TODO: ENV Var?
+MAX_FILE_SIZE = 10 * 1024 * 1024  # 10 MB
+
+os.makedirs(TEMP_DIR, exist_ok=True)
+
+
+def _convert_github_to_raw(github_url: str) -> str:
+    return (
+        github_url.replace("github.com", "raw.githubusercontent.com")
+        .replace("/blob/", "/")
+    )
+
+
+def _get_cb_info_from_repo(repo_url: str) -> Block:
+    if "github.com" in repo_url:
+        repo_url = _convert_github_to_raw(repo_url)
+
```

Comment:
Bad URL handling. Imagine: `https://time.rwth-aachen.de/github.com/` or smth like that. You should parse it, maybe there is a lib doing that for you.